src/createSchemaForm.tsx

```diff
@@ -430,29 +478,43 @@ export function createTsForm<
       };
       const ctxLabel = meta.description?.label;
       const ctxPlaceholder = meta.description?.placeholder;
-      accum[key] = (
-        <Fragment key={stringKey}>
+
+      return (
+        <Fragment key={prefixedKey}>
           {beforeElement}
           <FieldContextProvider
             control={control}
-            name={stringKey}
+            name={prefixedKey}
             label={ctxLabel}
             placeholder={ctxPlaceholder}
             enumValues={meta.enumValues as string[] | undefined}
             addToCoerceUndefined={addToCoerceUndefined}
             removeFromCoerceUndefined={removeFromCoerceUndefined}
           >
-            <Component key={key} {...mergedProps} />
+            <Component key={prefixedKey} {...mergedProps} />
           </FieldContextProvider>
           {afterElement}
         </Fragment>
       );
-      return accum;
-    }, {} as Record<SchemaKey, React.ReactNode>);
-    const renderedFieldNodes = Object.values(renderedFields);
+    }
+
+    function renderFields(schema: SchemaType, props: PropType<Mapping, SchemaType, PropsMapType> | undefined) {
+      type SchemaKey = keyof z.infer<UnwrapEffects<SchemaType>>;
+      const _schema = unwrapEffects(schema);
+      const shape: Record<string, RTFSupportedZodTypes> = _schema._def.shape();
+      return Object.entries(shape).reduce((accum, [key, type]: [SchemaKey, RTFSupportedZodTypes]) => {
+        // we know this is a string but TS thinks it can be number and symbol so just in case stringify
+        const stringKey = key.toString();
+        accum[stringKey] = renderComponentForSchemaDeep(type, props as any, stringKey, stringKey, getValues()[key]);
+        return accum;
+      }, {} as RenderedObjectElements) as RenderedFieldMap<SchemaType>;
```

Comment:
might be cool to get both the props and return values correct recursively but it was heinous and i'm not sure it gives us much